test/tests/functional/pbs_preemption.py

```diff
@@ -733,3 +733,57 @@ def test_preempt_requeue_resc(self):
 
         self.server.expect(JOB, {'job_state': 'R'}, id=hjid)
         self.server.expect(JOB, {'job_state=Q': 1})
+
+    def test_preempt_wrong_cull(self):
+        """
+        Test to make sure that if a preemptor cannot run because
+        it misses a non-consumable on a node, preemption candidates
+        are not incorrectly removed from consideration
+        if and because they "do not request the relevant resource".
+        Deciding on their utility should be left to the check
+        to see whether the nodes they occupy are useful.
+        """
+        attr = {}
+        attr['type'] = 'string_array'
+        attr['flag'] = 'h'
+
+        self.server.manager(MGR_CMD_CREATE, RSC, attr, id='app')
+        self.scheduler.add_resource('app')
+
+        a = {'resources_available.ncpus': 1, 'resources_available.app': 'appA'}
+        self.server.create_vnodes('vna', a, num=1, mom=self.mom,
+                                  usenatvnode=False)
+        b = {'resources_available.ncpus': 1, 'resources_available.app': 'appB'}
+        self.server.create_vnodes('vnb', b, num=1, mom=self.mom,
+                                  usenatvnode=False, additive=True)
+
+        # set the preempt_order to kill/requeue only -- try old and new syntax
+        self.server.manager(MGR_CMD_SET, SCHED, {'preempt_order': 'R'})
+
+        # create express queue
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "hipri")
+
+        # create normal queue
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 1}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "lopri")
+
+        # submit job 1
+        a = {'Resource_List.select': '1:ncpus=1:vnode=vna[0]', ATTR_q: 'lopri'}
+        j1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(j1)
+
+        self.server.expect(JOB, {'substate': '42'}, id=jid1)
```

Comment:
Don't worry about the substate, look for state: R.  PTL handles the substate for you. (here and below)

Reply:
Yes, indeed, I see it from the source (it adds substate 42 if you do not specify a substate explicitly). 

Changed to 'state': 'R'.